apps/examples/testcase/le_tc/network/itc_net_connect.c

```diff
@@ -0,0 +1,337 @@
+	address.sin_family = AF_INET;
+	address.sin_addr.s_addr = INADDR_ANY;
+	address.sin_port = htons(g_port);
+	ret = bind(server_socket, (struct sockaddr *)&address, sizeof(address));
+	if (ret != OK) {
+		close(server_socket);
+		*pret = ERROR;
+		pthread_exit(pret);
+	}
+	ret = listen(server_socket, 5);
+	if (ret != OK) {
+		close(server_socket);
+		*pret = ERROR;
+		pthread_exit(pret);
+	}
+	while (num_clients-- > 0) {
+		client_socket = accept(server_socket, (struct sockaddr *)&address, (socklen_t*)&addrlen);
+		if (client_socket < 0) {
+			close(server_socket);
+			*pret = ERROR;
+			pthread_exit(pret);
+		}
+		total_recv = strlen(CLIENT_MSG);
+		memset(buffer, 0, BUFF_LEN);
+		while (total_recv) {
+			valrecv = recv(client_socket, buffer + recv_len, BUFF_LEN - recv_len, MSG_WAITALL);
+			printf("server recv: %s\n", buffer);
+			if (valrecv == -1) {
+				close(client_socket);
+				close(server_socket);
+				pthread_exit(pret);
+			}
+			total_recv -= valrecv;
+			if (total_recv < 0) {
+				close(client_socket);
+				close(server_socket);
+				*pret = ERROR;
+				pthread_exit(pret);
+			}
+			recv_len += valrecv;
+		}
+		strncpy(buffer + recv_len, ptr_msg, sizeof(SERVER_MSG));
+		total_send = strlen(buffer);
+		while (total_send) {
+			valsend = send(client_socket, buffer + send_len, strlen(buffer) - send_len, 0);
+			printf("server send: %s\n", buffer);
+			if (valsend == -1) {
+				close(client_socket);
+				close(server_socket);
+				*pret = ERROR;
+				pthread_exit(pret);
+			}
+			total_send -= valsend;
+			if (total_send < 0) {
+				close(client_socket);
+				close(server_socket);
+				*pret = ERROR;
+				pthread_exit(pret);
+			}
+			send_len += valsend;
+		}
+		sleep(SERVER_WAIT_TIME);
+		ret = close(client_socket);
+		if (ret != OK) {
+			close(server_socket);
+			*pret = ERROR;
+			pthread_exit(pret);
+		}
+		printf("Server thread : finished with client\n");
+	}
+	ret = close(server_socket);
+	if (ret != OK) {
+		*pret = ERROR;
+		pthread_exit(pret);
+	}
+	printf("Server completed\n");
+	pthread_exit(pret);
+}
+
+/**
+* @fn               :client_connect
+* @description      :starts client
+* @return           :void
+*/
+static void *client_connect(void *ptr_id)
+{
+	int *id;
+	id = (int *)ptr_id;
+	int sock = 0;
+	int ret;
+	int valrecv = 0;
+	int valsend = 0;
+	struct sockaddr_in serv_addr;
+	char client_msg[BUFF_LEN];
+	char buffer[BUFF_LEN] = {0};
+	int total_recv;
+	int total_send;
+	int *pret = malloc(sizeof(int));
+	int recv_len = 0;
+	int send_len = 0;
+	*pret = OK;
+	printf("Client thread started...\n");
+	sock = socket(AF_INET, SOCK_STREAM, 0);
+	if (sock < 0) {
+		pthread_exit(pret);
+	}
+	memset(&serv_addr, '0', sizeof(serv_addr));
+	serv_addr.sin_family = AF_INET;
+	serv_addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+	serv_addr.sin_port = htons(g_port);
+	ret = connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
+	if (ret != OK) {
+		close(sock);
+		return (void *)ret;
```

Comment:
Here is return, not pthread_exit. If you use return, pret causes memory leak.

Reply:
changes are done as per your suggestion. please review